pkg/app/piped/cmd/piped/piped_test.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package piped
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHasTooManyConfigFlags(t *testing.T) {
+	t.Parallel()
+
+	testcases := []struct {
+		title     string
+		p         *piped
+		expectErr bool
+	}{
+		{
+			title: "no config",
+			p: &piped{
+				configFile:      "",
+				configGCPSecret: "",
+				configAWSSecret: "",
+			},
+			expectErr: false,
+		},
+		{
+			title: "only one config is set",
+			p: &piped{
+				configFile:      "config.yaml",
+				configGCPSecret: "",
+				configAWSSecret: "",
+			},
+			expectErr: false,
+		},
+		{
+			title: "two configs are set",
+			p: &piped{
+				configFile:      "config.yaml",
+				configGCPSecret: "xxx",
+				configAWSSecret: "",
+			},
+			expectErr: true,
+		},
+	}
```

Comment:
[nit] add a case for all of them are set 👀

Reply:
@ffjlabo Thanks!! I added a case of three.